src/fides/api/models/location_regulation_selections.py

```diff
@@ -195,3 +257,18 @@ async def get_selected_regulations_async(
             if record:
                 return set(record.selected_regulations)
             return set()
+
+
+def group_locations_into_location_groups(saved_locations: Iterable[str]) -> Set[str]:
+    """Combines saved Locations into Location groups if applicable
+    All Locations must be present for a Location Group to be valid.
+    """
+    from fides.api.schemas.locations import location_group_to_location
```

Comment:
oof i feel like i've set us up with a bad structure between the db model and the API schema to get us to the point where we need to do this 🤔 

i can see it's not straightforward to resolve this dep conflict so i get why you did this. the best resolution/alternative i see at the moment - i feel we'll probably set ourselves up better moving forward if we move a lot of the enum creation and baseline location "registry" loading out of the API schema and either into the DB model class or into a shared util that both can depend on/point to. we don't need to do this in the current iteration since this should be fine (not trying to make more work that's not strictly necessary!)

i just teased this out a bit in my head, haven't explored thoroughly or worked out fully how it fits together with your updates across both repos. again, we don't need to do this now, especially if it doesn't feel straightforward to you. but whenever we need to do an in-function import like this, it gives me pause and makes me concerned about maintainability...

Reply:
yep, I was taking a shortcut here - since you flagged as well, I can spend a little time trying to untangle.

Reply:
For now it's on the model! We can figure out how to extract better later.  If I didn't have location groups being a side effect of saving locations this would be easier but I know it was desired to group this action as much as possible.